### ISSUE TITLE
GET with a \u0000 seems to cause exception if Static middleware is used

### ISSUE THREAD

**Issue by grepsuzette** [kevinresol/tink_http_middleware#5]
(Slowly starting to run some web app pentest tools againt tink_web)

There is a possible duplicate of https://github.com/haxetink/tink_web/issues/107 (but I think not, this one happens in tink.web.middleware.StaticHandler)

Some stupid GET request containing a unicode `\u0000`.

```
TypeError [ERR_INVALID_ARG_VALUE]: The argument 'path' must be a string or Uint8Array without null bytes. Received '/path/to/the/server/www/index.html/web-console/ServerInfo.jsp\u0000'
    at Object.stat (fs.js:1001:10)
    at Function.asys_FileSystem.exists (/path/to/the/server/server.js:3918:13)
    at tink_http_middleware_StaticHandler.process (/path/to/the/server/server.js:12006:154)
    at tink_http_middleware_StaticHandler.process (/path/to/the/server/server.js:12043:23)
    at Server.<anonymous> (/path/to/the/server/server.js:11578:11)
    at Server.emit (events.js:315:20)
    at parserOnIncoming (_http_server.js:779:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:117:17) {
  code: 'ERR_INVALID_ARG_VALUE'
}
```

Will try to fix it.

**Comment by kevinresol** [comment]
(Transferred to here)

**Comment by kevinresol** [comment]
I think some sanitization/prechecks should be done here:

https://github.com/haxetink/tink_http_middleware/blob/f2b8010cbb8e37080afa3dd3163ee61c5f1fee04/src/tink/http/middleware/Static.hx#L72

**Comment by grepsuzette** [comment]
I'm in the file but since `asys` uses tink, I wonder if the problem should not be fixed there. As soon as we are in tinkland we should let no exception.

**Comment by grepsuzette** [comment]
~After studying it think it should not be fixed neither in middleware nor in asys, but returning an http error even before middlewares (if that's possible). see https://github.com/haxetink/tink_web/issues/107~

Edit: Nope should be done here and in every middleware that may have the same pb.

**Comment by back2dos** [comment]
Fixed via #6.
